### DISCUSSION TITLE
Can't access JPCert LogonTracer docker image on Security Onion

### DISCUSSION THREAD

**Discussion by UMWP** [Security-Onion-Solutions/securityonion#6592] [Unsupported Versions]
SO 2.3.90 in standalone mode.

I'm trying to install jpcert logontracer:
https://github.com/JPCERTCC/LogonTracer/wiki/jump-start-with-docker 

After successfully pulling docker image I start it with following command (changed default port from 8080 to 8443 because default port is already used by osquery):
c@soc ~]$ sudo docker run --detach --publish=7474:7474 --publish=7687:7687 --publish=8443:8443 -e LTHOSTNAME=10.1.1.117 jpcertcc/docker-logontracer
ce73c3714129623776bc43229cda8f62665eed7c69ef02896d6be4374694f9a6

I have also added port 8443 to default port group according to documentation 
https://docs.securityonion.net/en/2.3/firewall.html#modify-a-default-port-group

I have also checked
c@soc ~]$ sudo iptables -vnL|grep 8443
    0     0 ACCEPT     tcp  --  !docker0 docker0  0.0.0.0/0            172.17.0.31          tcp dpt:8443
   40  2080 ACCEPT     tcp  --  *      *       10.1.1.0/24          0.0.0.0/0            tcp dpt:8443
    0     0 ACCEPT     tcp  --  *      *       127.0.0.1            0.0.0.0/0            tcp dpt:8443

Unfortunatelly I still can't get to http://10.1.1.117:8443

What could be wrong? I have no experience with docker, so forgive me any obvious/trivial mistakes.

**Comment by tomjones1977** [comment]
Is it definitely 8443 in the container?  The getting started docs on the logontracer github page mention port 8080 rather than 8443.

**Comment by UMWP** [comment]
As as I've described when I tried to run it on default port 8080 it stated that it is already used (I suppose by osquery), so I switched to 8443.
Anyway thank You for your fast response.

**Reply by tomjones1977** [reply]
While you have changed the listening port of the container, you haven't changed what the application inside the container is doing.  That will still listen on 8443.  One way around this would be to change your expose line to `8443:8080`.  This will map connections to the container on 8443 to 8080 inside the container.

**Comment by UMWP** [comment]
By the way, port 8080 is used by fleet container.
Theoretically it should be OK:
[soc@soc ~]$ sudo docker ps|grep 8080
cb390703fa38   jpcertcc/docker-logontracer                                      "/sbin/tini -g -- /d…"   2 minutes ago   Up 2 minutes           0.0.0.0:7474->7474/tcp, 7473/tcp, 0.0.0.0:7687->7687/tcp, 0.0.0.0:8443->8080/tcp                                                   zealous_hypatia
4dba6f494f4c   soc:5000/security-onion-solutions/so-fleet:2.3.90                "/startfleet.sh"         8 days ago      Up 6 hours             0.0.0.0:8080->8080/tcp                                                                                                             so-fleet

But I still can't get to http://10.1.1117:8443  :-(

I get no answer from that port:
[soc@soc ~]$ sudo tcpdump -i ens192 -nn port 8443
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on ens192, link-type EN10MB (Ethernet), capture size 262144 bytes
14:15:35.143982 IP 10.1.1.100.58376 > 10.1.1.117.8443: Flags [S], seq 572616671, win 64240, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
14:15:35.144501 IP 10.1.1.100.58377 > 10.1.1.117.8443: Flags [S], seq 1623610756, win 64240, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
14:15:35.397338 IP 10.1.1.100.58378 > 10.1.1.117.8443: Flags [S], seq 943069324, win 64240, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0

Any ideas?

**Reply by tomjones1977** [reply]
That's fine, as each container has it's own internal networking, but they can't share an external port on the host.  For example, you could have three web servers, in three containers, all running on port 80.  On the host you would map 80 to container 1, 81 to container 2, and 82 to container 3.  That works as they are all in their own container.

**Comment by UMWP** [comment]
If container configuration (port->port mapping) is OK it seems for me like SO firewall has something with it
I have done exactly what they say here:
https://docs.securityonion.net/en/2.3/firewall.html#modify-a-default-port-group
but I used port 8443 instead of described in example port 8086

By command: 
sudo so-firewall addportgroup nginx
it complained that it already exist

So I entered only:
sudo so-firewall addport nginx tcp 8443

and the rest according to description.

**Reply by tomjones1977** [reply]
What is in the firewall section of your `/opt/so/saltstack/local/pillar/minions/<HOSTNAME>_<ROLE>.sls` file?

**Reply by UMWP** [reply]
firewall:
  assigned_hostgroups:
    chain:
      DOCKER-USER:
        hostgroups:
          analyst:
            portgroups:
              - portgroups.nginx

**Comment by weslambert** [comment]
Instead of trying to expose the application port externally, I'd recommend you proxy it with nginx, using a subpath like `/logontracer/`:

![image](https://user-images.githubusercontent.com/16829864/146565191-e93e2eca-37c1-487b-ae59-c2675e66340d.png)

Please keep in mind, this kind of integration is not officially supported. We don't officially support installing and managing external/non-SO components, so beware of unintended erasure, overrides, and/or service restarts.

**Reply by UMWP** [reply]
Sorry about late answer and thank You for interesting idea. Unfortunately still something wrong. :-(

I start logontracer docker container like this:
docker run --detach --publish=7474:7474 --publish=7687:7687 --publish=8443:8443 -e LTHOSTNAME=10.1.1.117 jpcertcc/docker-logontracer

Do You mean to add in /opt/so/saltstack/default/salt/nginx/etc/nginx.conf following lines:
location /logontracer/ {
                             if ($http_authorization = "") {
                                     return 403;
                             }
                             proxy_pass            http://{{ manager_ip }}:8443/;
                             proxy_read_timeout    90;
                             proxy_connect_timeout 90;
                             proxy_set_header      x-user-id "";
                             proxy_set_header      Host $host;
                             proxy_set_header      X-Real-IP $remote_addr;
                             proxy_set_header      X-Forwarded-For $proxy_add        _x_forwarded_for;
                             proxy_set_header      Proxy "";
                             proxy_set_header      X-Forwarded-Proto $scheme;
                     }

and then restart sudo so-nginx-restart

I did it and I got 403 Forbidden :-(

Any hints?

P.S.
Don't worry about support - it's only lab right now.

**Reply by UMWP** [reply]
It's a little off topic, by my humble proposition is to think about adding RITA (Real Intelligence Threat Analytics) to next official SO release as optional module to choose during installation phase.
